### PR TITLE
ls: New output format when versioning is enabled

### PR DESCRIPTION
**Pull request by vadmeste**
Below is the new listing output:

Screenshot from 2020-09-01 12-15-54

### CODE REVIEW

**Comment by harshavardhana** [comment]
Now should be New?

**Comment by vadmeste** [comment]
> Now should be New?

yes, fixed

**Review by kannappanr — APPROVED** [review]
LGTM

**Comment by kannappanr** [comment]
@vadmeste Please resolve conflicts

**Comment by vadmeste** [comment]
> @vadmeste Please resolve conflicts

Done

File: cmd/ls.go [hunk 1]
```diff
@@ -56,20 +56,23 @@ type contentMessage struct {
 func (c contentMessage) String() string {
 	message := console.Colorize("Time", fmt.Sprintf("[%s] ", c.Time.Format(printDate)))
 	message += console.Colorize("Size", fmt.Sprintf("%7s ", strings.Join(strings.Fields(humanize.IBytes(uint64(c.Size))), "")))
-	fileDesc := c.Key
+	if c.Filetype == "folder" {
+		return message + console.Colorize("Dir", c.Key)
```
Comment (harshavardhana) [inline cmd/ls.go:60]: We shouldn't return here @vadmeste make sure that we print versions for directories as well - look at the changes present in the master branch - adopt this accordingly.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/ls.go (hunk 1 above)
Comment (vadmeste) [reply cmd/ls.go:60]: > make sure that we print versions for directories

Not sure I understand, directories do not have versions in S3 spec AFAIK, could you explain a little bit more ?

Comment (harshavardhana) [reply cmd/ls.go:60]: Directories are objects and are indeed versioned @vadmeste 

```
~ mc ls -r --versions s3/kannappannvme/
[2019-06-02 20:17:42 PDT]     99B dd_tests/dd-r-drive1-4M-256.in (1st, vid=null)
[2019-06-02 20:17:42 PDT]     99B dd_tests/dd-r-drive2-4M-256.in (1st, vid=null)
[2019-06-02 20:17:42 PDT]     99B dd_tests/dd-r-drive3-4M-256.in (1st, vid=null)
[2019-06-02 20:17:42 PDT]     99B dd_tests/dd-r-drive4-4M-256.in (1st, vid=null)
[2019-06-02 20:17:42 PDT]     99B dd_tests/dd-r-drive5-4M-256.in (1st, vid=null)
[2019-06-02 20:17:42 PDT]     99B dd_tests/dd-r-drive6-4M-256.in (1st, vid=null)
[2019-06-02 20:17:42 PDT]     98B dd_tests/dd-r-drive7-4M-256.in (1st, vid=null)
[2019-06-02 20:17:42 PDT]     98B dd_tests/dd-r-drive8-4M-256.in (1st, vid=null)
[2019-06-02 20:17:42 PDT]     99B dd_tests/dd-w-drive1-4M-256.out (1st, vid=null)
[2019-06-02 20:17:42 PDT]     99B dd_tests/dd-w-drive2-4M-256.out (1st, vid=null)
[2019-06-02 20:17:42 PDT]     99B dd_tests/dd-w-drive3-4M-256.out (1st, vid=null)
[2019-06-02 20:17:42 PDT]     99B dd_tests/dd-w-drive4-4M-256.out (1st, vid=null)
[2019-06-02 20:17:42 PDT]     99B dd_tests/dd-w-drive5-4M-256.out (1st, vid=null)
[2019-06-02 20:17:42 PDT]     99B dd_tests/dd-w-drive6-4M-256.out (1st, vid=null)
[2019-06-02 20:17:42 PDT]     99B dd_tests/dd-w-drive7-4M-256.out (1st, vid=null)
[2019-06-02 20:17:42 PDT]     99B dd_tests/dd-w-drive8-4M-256.out (1st, vid=null)
[2019-06-02 20:17:42 PDT]    251B dd_tests/dd.sh (1st, vid=null)
[2019-06-02 20:17:42 PDT]  5.5KiB iozone_tests/multiple-drives-16-threads (1st, vid=null)
[2019-06-02 20:17:42 PDT]  5.7KiB iozone_tests/multiple-drives-32-threads (1st, vid=null)
[2019-06-02 20:17:42 PDT]  6.2KiB iozone_tests/multiple-drives-64-threads (1st, vid=null)
[2019-06-02 20:17:42 PDT]  5.3KiB iozone_tests/multiple-drives-8-threads (1st, vid=null)
[2019-06-02 20:17:42 PDT]  5.5KiB iozone_tests/single-drive-16-threads (1st, vid=null)
[2019-06-02 20:17:43 PDT]  5.3KiB iozone_tests/single-drive-2-threads (1st, vid=null)
[2019-06-02 20:17:43 PDT]  5.8KiB iozone_tests/single-drive-32-threads (1st, vid=null)
[2019-06-02 20:17:43 PDT]  5.3KiB iozone_tests/single-drive-4-threads (1st, vid=null)
[2019-06-02 20:17:43 PDT]  6.2KiB iozone_tests/single-drive-64-threads (1st, vid=null)
[2019-06-02 20:17:43 PDT]  5.4KiB iozone_tests/single-drive-8-threads (1st, vid=null)
[2019-06-02 20:17:43 PDT]  5.2KiB iozone_tests/single-drive-single-thread (1st, vid=null)
[2020-09-17 15:16:54 PDT]      0B ppp/ (1st, vid=null)
[2020-09-18 13:44:05 PDT]      0B prefix/ (4th, vid=iGGl3Y4l5WoZBaS48SXOCPw8BzPgCSw.)
[2020-09-18 13:44:04 PDT]      0B prefix/ (3rd, vid=LvwrzCLqLma8IElk_hDVjlmi5m9dkro.)
[2020-09-18 13:44:03 PDT]      0B prefix/ (2nd, vid=INeQLQ3TpxjVacTBinie2FhGeHu8Ykzb)
[2020-09-18 13:43:30 PDT]      0B prefix/ (1st, vid=null)
[2020-09-16 21:29:14 PDT]     26B prefix/issue (1st, vid=null)
```

Please keep the support that I added in master and adopt this change accordingly @vadmeste

Comment (vadmeste) [reply cmd/ls.go]: > We shouldn't return here @vadmeste make sure that we print versions for directories as well - look at the changes present in the master branch - adopt this accordingly.

Done anyway

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**